### PR TITLE
dark-mode-notify: init at 2022-07-18

### PR DESCRIPTION
**Pull request by YorikSar**
###### Description of changes

`dark-mode-notify` is a small too that calls a binary whenever dark mode status changes in macOS.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -18049,6 +18049,14 @@
     githubId = 647076;
     name = "Yorick van Pelt";
   };
+  YorikSar = {
+    name = "Yuriy Taraday";
+    email = "yorik.sar@gmail.com";
+
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:18055]:
```suggestion
```

File: pkgs/os-specific/darwin/dark-mode-notify/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+, fetchFromGitHub
+, stdenv
+, swift
+, swiftpm
+, darwin
+}:
+
+stdenv.mkDerivation (final: {
+  pname = "dark-mode-notify";
+  version = "2022-07-18";
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/dark-mode-notify/default.nix:11]:
```suggestion
  version = "unstable-2022-07-18";
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (YorikSar) [reply maintainers/maintainer-list.nix:18055]: Applied.

File: pkgs/os-specific/darwin/dark-mode-notify/default.nix (hunk 2 above)
Comment (YorikSar) [reply pkgs/os-specific/darwin/dark-mode-notify/default.nix:11]: Applied.

File: maintainers/maintainer-list.nix [hunk 3]
```diff
@@ -1,61 +1,61 @@
...
-    handle = {
-      # Required
-      name = "Your name";
+  ```nix
+  handle = {
+  # Required
+  name = "Your name";
 
-      # Optional, but at least one of email, matrix or githubId must be given
-      email = "address@example.org";
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:8]: Please revert all the formatting changes done in this file

File: pkgs/os-specific/darwin/dark-mode-notify/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+, darwin
+}:
+
+stdenv.mkDerivation (final: {
+  pname = "dark-mode-notify";
+  version = "unstable-2022-07-18";
+
+  src = fetchFromGitHub {
+    owner = "bouk";
+    repo = final.pname;
```
Comment (SuperSandro2000) [inline pkgs/os-specific/darwin/dark-mode-notify/default.nix:15]:
```suggestion
    repo = "dark-mode-notify";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/darwin/dark-mode-notify/default.nix (hunk 4 above)
Comment (YorikSar) [reply pkgs/os-specific/darwin/dark-mode-notify/default.nix:15]: Applied

**Review by infinisil — APPROVED** [review]
LGTM!

**Pull request merged**